scope_registry.go

```diff
@@ -125,11 +125,11 @@ func (r *scopeRegistry) CachedReport() {
 
 func (r *scopeRegistry) ForEachScope(f func(*scope)) {
 	for _, subscopeBucket := range r.subscopes {
```

Comment:
Quick sanity check: r.subscopes is immutable after construction, right?

Reply:
yes, the buckets in r.subcsopes remain the same. though buckets will have subscopes added to them when subscope() is called.